host/class/uvc/usb_host_uvc_2/examples/basic_uvc_stream/main/basic_uvc_stream.c

```diff
@@ -0,0 +1,313 @@
+        ESP_LOGI(TAG, "Device 0x%04X:0x%04X-%d OPENED!", stream_config->usb.vid, stream_config->usb.pid, uvc_index);
+        vTaskDelay(pdMS_TO_TICKS(100));
+        unsigned count = 0;
+
+        // This is the main processing loop. It enables the stream for 2 seconds and then closes it
+        uvc_host_stream_start(uvc_stream);
+        while (true) {
+            ESP_LOGI(TAG, "Stream %d start. Iteration %u", uvc_index, count);
+            count++;
+            TickType_t timeout_ticks = pdMS_TO_TICKS(EXAMPLE_RECORDING_LENGTH_S * 1000);
+            TimeOut_t connection_timeout;
+            vTaskSetTimeOutState(&connection_timeout);
+
+            uvc_host_stream_unpause(uvc_stream);
+            do {
+                uvc_host_frame_t *frame;
+                if (xQueueReceive(frame_q, &frame, pdMS_TO_TICKS(1000)) == pdPASS) {
+                    ESP_LOGI(TAG, "Stream %d: New frame! Len: %d", uvc_index, frame->data_len);
+
+                    // Process the frame data here!!
+
+                    uvc_host_frame_return(uvc_stream, frame);
+                } else {
+                    ESP_LOGW(TAG, "Stream %d: Frame not received on time", uvc_index);
+                }
+            } while (xTaskCheckForTimeOut(&connection_timeout, &timeout_ticks) == pdFALSE);
+            ESP_LOGI(TAG, "Stream %d stop", uvc_index);
+            uvc_host_stream_pause(uvc_stream);
+
+            vTaskDelay(pdMS_TO_TICKS(2000));
+        }
+        uvc_host_stream_stop(uvc_stream);
+        uvc_host_stream_close(uvc_stream);
+
+        // We are done. Wait for device disconnection and start over
+        ESP_LOGI(TAG, "Example finished successfully! You can reconnect the device to run again.");
+        xSemaphoreTake(device_disconnected_sem, portMAX_DELAY);
+    }
+}
+
+static const uvc_host_stream_config_t stream_mjpeg_config = {
+    .event_cb = handle_event,
+    .frame_cb = frame_callback,
+    .user_ctx = &rx_frames_queue[0],
+    .usb.vid = EXAMPLE_USB_DEVICE_VID,
+    .usb.pid = EXAMPLE_USB_DEVICE_PID,
+    .usb.uvc_function_index = 0,
+    .vs_format.h_res = 720,
+    .vs_format.v_res = 1280,
+    .vs_format.fps = 15,
+    .vs_format.format = UVC_VS_FORMAT_MJPEG,
+    .advanced.number_of_frame_buffers = EXAMPLE_FRAME_COUNT,
+    .advanced.frame_size = 0,
+    .advanced.number_of_urbs = 6,
+    .advanced.urb_size = 20 * 1024,
+};
+
+static const uvc_host_stream_config_t stream_h265_config = {
+    .event_cb = handle_event,
+    .frame_cb = frame_callback,
+    .user_ctx = &rx_frames_queue[1],
+    .usb.vid = EXAMPLE_USB_DEVICE_VID,
+    .usb.pid = EXAMPLE_USB_DEVICE_PID, // Customer's device
+    .usb.uvc_function_index = 1,
+    .vs_format.h_res = 1280,
+    .vs_format.v_res = 720,
+    .vs_format.fps = 15,
+    .vs_format.format = UVC_VS_FORMAT_H265,
+    .advanced.number_of_frame_buffers = EXAMPLE_FRAME_COUNT,
+    .advanced.frame_size = 0,
+    .advanced.number_of_urbs = 6,
+    .advanced.urb_size = 20 * 1024,
+};
+
+/*
```

Comment:
This function can be used for initializing SD card and saving the frames on it. It can be useful during testing.

I'll clean the example up before releasing it, thanks!